lib/cli/src/latest_version.js

```diff
@@ -122,29 +122,25 @@ function spawnVersionsWithYarn(packageName, constraint) {
  * @param {Object} constraint Version range to use to constraint the returned version
  * @returns {Promise<string|Array<string>>} versions  Promise resolved with a version or an array of versions
  */
-function spawnVersionsWithYarn2(packageName, constraint) {
+async function spawnVersionsWithYarn2(packageName, constraint) {
```

Comment:
can we update the calling code to use async/await rather than handling promises directly? i know it's basically the same thing, but easier to follow. my first instinct when reviewing this was "doesn't he need to update the calling code?"

Reply:
✅ Done @shilman, `deepscan` is finding a new issue but it's due to the way the code is formatted by prettier 🤷

Also, I plan to do a major refactoring to split the custom behavior of Yarn, Yarn 2, and NPM in their own files and use a kind of factory to instantiate them, instead of having to check which package manger should be used in time we need to run a command. But now that we are in `beta` I should probably wait for 6.1 no?

Reply:
Thanks @gaetanmaisse ! Beta means that we don't intend to make destabilizing changes between now and release. If it's mostly moving code around to clean things up, I think that's fine.